### PR TITLE
feat(provider/kubernetes): add dns policy

### PR DESCRIPTION
**Pull request by ethanfrogers**
adds support for dns policy on server groups and run job stages.

https://kubernetes.io/docs/api-reference/v1.7/#pod-v1-core

Corresponding Clouddriver PR: spinnaker/clouddriver#1800

@danielpeach PTAL

### CODE REVIEW

File: app/scripts/modules/kubernetes/pipeline/stages/runJob/runJobStage.html [hunk 1]
```diff
@@ -30,6 +30,17 @@
     </div>
   </stage-config-field>
 
+  <stage-config-field label="DNS Policy">
+    <div>
+      <select class="form-control input-sm"
+                ng-model="ctrl.stage.dnsPolicy">
```
Comment (danielpeach) [inline app/scripts/modules/kubernetes/pipeline/stages/runJob/runJobStage.html]: Please align this attribute with the one above it.

File: app/scripts/modules/kubernetes/pipeline/stages/runJob/runJobStage.html [hunk 2]
```diff
@@ -30,6 +30,17 @@
...
+          <option ng-repeat="policy in ctrl.policies"
```
Comment (danielpeach) [inline app/scripts/modules/kubernetes/pipeline/stages/runJob/runJobStage.html]: There's another angular syntax for `<select>`s that's a little bit cleaner (I think) and lets you omit `<option>`: https://github.com/spinnaker/deck/blob/master/app/scripts/modules/amazon/src/serverGroup/configure/wizard/zones/availabilityZoneSelector.directive.html#L7-L9

But I don't remember the differences between the two and there may be some good reason for the syntax you've used.

File: app/scripts/modules/kubernetes/pipeline/stages/runJob/runJobStage.js [hunk 3]
```diff
@@ -55,6 +57,9 @@ module.exports = angular.module('spinnaker.kubernetes.pipeline.stage.runJobStage
       _.set(this.stage, 'container.imageDescription.fromTrigger', false);
     }
 
+    if (!_.has(this.stage, 'dnsPolicy')) {
```
Comment (danielpeach) [inline app/scripts/modules/kubernetes/pipeline/stages/runJob/runJobStage.js]: If we're sure `this.stage` exists, it's cleaner to omit the `has`/`set` here. 
e.g.,
```
if (!this.stage.dnsPolicy) {
  this.stage.dnsPolicy = 'ClusterFirst';
}
```

File: app/scripts/modules/kubernetes/serverGroup/configure/wizard/advancedSettings.controller.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+'use strict';
+
+import _ from 'lodash';
+const angular = require('angular');
+
+module.exports = angular.module('spinnaker.serverGroup.configure.kubernetes.advancedSettings', [])
+  .controller('kubernetesServerGroupAdvancedSettingsController', function($scope) {
+
+    if (!_.has($scope.command, 'dnsPolicy')) {
```
Comment (danielpeach) [inline app/scripts/modules/kubernetes/serverGroup/configure/wizard/advancedSettings.controller.js]: Same here - `$scope.command` should definitely exist.

File: app/scripts/modules/kubernetes/serverGroup/configure/wizard/advancedSettings.html [hunk 5]
```diff
@@ -1,6 +1,23 @@
...
+        <help-field key="kubernetes.pod.dnsPolicy"></help-field>
+      </div>
+      <div class="col-md-7">
+        <select class="form-control input-sm"
+                ng-model="command.dnsPolicy">
+          <option ng-repeat="policy in advancedController.policies"
+                  value="{{policy}}"
+                  ng-selected="command.dnsPolicy === policy">{{policy}}</option>
+        </select>
+
```
Comment (danielpeach) [inline app/scripts/modules/kubernetes/serverGroup/configure/wizard/advancedSettings.html]: Extra whitespace.

File: app/scripts/modules/kubernetes/pipeline/stages/runJob/runJobStage.html (hunk 1 above)
Comment (ethanfrogers) [inline app/scripts/modules/kubernetes/pipeline/stages/runJob/runJobStage.html]: One of these days I won't mess this up! 😭

File: app/scripts/modules/kubernetes/pipeline/stages/runJob/runJobStage.html (hunk 2 above)
Comment (ethanfrogers) [inline app/scripts/modules/kubernetes/pipeline/stages/runJob/runJobStage.html]: TBH, I'm not familiar enough with Angular to have an opinion either way.

**Comment by ethanfrogers** [comment]
@danielpeach comments addresses. thanks!

**Pull request merged**